### PR TITLE
Wazo 2764 stasis unsubscribe

### PR DESCRIPTION
**Pull request by pc-m**

### CODE REVIEW

File: resource_amqp.h [hunk 1]
```diff
@@ -61,5 +59,15 @@ struct ast_ari_amqp_stasis_unsubscribe_args {
 	/*! Application's name */
 	const char *application_name;
 };
+/*!
+ * \brief Delete a stasis subscription to AMQP.
+ *
+ * Delete a Stasis application and subscribe to it's event and forward them to AMQP. The application's name must be unique.
```
Comment (bloom1) [inline resource_amqp.h:65]: Nitpicking: `it's event` => `its events`

File: amqp.json [hunk 2]
```diff
@@ -33,6 +33,33 @@
 							"reason": "Bad request."
 						}
 					]
+				},
+				{
+					"httpMethod": "DELETE",
+					"summary": "Delete a stasis subscription to AMQP.",
+					"notes": "Delete a Stasis application and subscribe to it's event and forward them to AMQP. The application's name must be unique.",
```
Comment (bloom1) [inline amqp.json:40]: Nitpicking: `it's event` => `its events`

File: res_ari_amqp.c [hunk 3]
```diff
@@ -1,9 +1,9 @@
 /*
  * Asterisk -- An open source telephony toolkit.
  *
- * Copyright (C) 2019 The Wazo Authors  (see the AUTHORS file)
+ * Copyright (C) 2019-2022 The Wazo Authors  (see the AUTHORS file)
  *
- * Nicolaos Ballas
+ * Wazo
```
Comment (bloom1) [inline res_ari_amqp.c:6]: Isn't that redundant with the copyright line?

File: resource_amqp.c [hunk 4]
```diff
@@ -1,9 +1,9 @@
 /*
  * Asterisk -- An open source telephony toolkit.
  *
- * Copyright (C) 2019 The Wazo Authors  (see the AUTHORS file)
+ * Copyright (C) 2019-2022 The Wazo Authors  (see the AUTHORS file)
  *
- * Nicolaos Ballas
+ * Wazo
```
Comment (bloom1) [inline resource_amqp.c:6]: Same as previous comment about copyright

File: resource_amqp.h [hunk 5]
```diff
@@ -1,9 +1,9 @@
 /*
  * Asterisk -- An open source telephony toolkit.
  *
- * Copyright (C) 2019 The Wazo Authors  (see the AUTHORS file)
+ * Copyright (C) 2019-2022 The Wazo Authors  (see the AUTHORS file)
  *
- * Nicolaos Ballas
+ * Wazo
```
Comment (bloom1) [inline resource_amqp.h:6]: Same as previous comment about copyright

File: res_ari_amqp.c [hunk 6]
```diff
@@ -105,6 +107,67 @@ static void ast_ari_amqp_stasis_subscribe_cb(
 	}
 #endif /* AST_DEVMODE */
 
+fin: __attribute__((unused))
+	return;
```
Comment (bloom1) [inline res_ari_amqp.c:111]: What is the purpose of this?

**Review by bloom1 — CHANGES_REQUESTED** [review]

File: res_ari_amqp.c (hunk 3 above)
Comment (pc-m) [reply res_ari_amqp.c:6]: It's auto-generated code from the template which uses the `_author` field

File: res_ari_amqp.c (hunk 6 above)
Comment (pc-m) [reply res_ari_amqp.c:111]: The `fin` label is auto-generated from the template. the `__attribute__((unused))` prevents the compiler from generating a warning when the label is unused.

Comment (bloom1) [reply res_ari_amqp.c:111]: Thank you!

**Review by bloom1 — APPROVED** [review]

**Pull request merged**